zerver/lib/test_classes.py

```diff
@@ -652,6 +653,9 @@ def mit_email(self, name: str) -> str:
     def notification_bot(self, realm: Realm) -> UserProfile:
         return get_system_bot(settings.NOTIFICATION_BOT, realm.id)
 
+    def refresh_user(self, user: UserProfile) -> UserProfile:
+        return get_user_profile_by_id(user.id)
```

Comment:
This might be a function where we want to clearly define via a docstring its semantics in relation to using a cache vs going to the database.

Reply:
I'm fine with that.  I also want to consider other names like the following:

* refreshed_user
* refetched_user
* fetch_user_again
* reload_user_using_app

I don't think any of the above are necessarily superior to what I chose, but there may be a better name, for sure.

Reply:
I'm pretty happy with `refresh_user` or `refetch_user`.

I feel like in a way the ideal code here might be to fetch the user from both cache and database and confirm the results are the same, if it were cheap to do so, but I'm not at all convinced it is cheap (from a code to write perspective; performance should be fine, as it's like 50 calls total!)... mainly because the code paths that we're using this in are ones where someone was worried about the test not testing what is expected due to a stale object.